sql/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java

```diff
@@ -1966,4 +1969,31 @@ public void testCastTimestampWithoutTimeZoneFromStringLiteralUsingSQLStandardFor
         AnalyzedRelation relation = analyze("select timestamp without time zone '2018-12-12 00:00:00'");
         assertThat(relation.outputs().get(0).valueType(), is(DataTypes.TIMESTAMP));
     }
+
+    @Test
+    public void test_element_within_object_array_of_derived_table_can_be_accessed_using_subscript() {
+        analyze("select s.friends['id'] from (select friends from doc.users) s");
```

Comment:
should we add some basic assertions to these tests that the selected columns in the top level relation are correctly resolved?